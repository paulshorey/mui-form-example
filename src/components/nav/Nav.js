// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import NavGroup from './NavGroup';
import * as Styled from './NavStyled';
import * as uiActions from 'redux/actions/ui';

type Props = {
	routes: [], // list of routes/urls/components from src/devices
	ui: {}, // window.store.ui from redux
};

export class Nav extends Component<Props> {
	renderLinks = (links: any) => {
		return links.map((link, index) => {
			return (
				<NavLink
					key={link.url + index}
					className="title link"
					activeClassName="active"
					to={link.url}
					onClick={() => {
						this.props.dispatch(uiActions.UI_NAV_CLOSE());
					}}
				>
					<span className="fontIcon icon-navlink_dot" />
					<span>{link.title}</span>
				</NavLink>
			);
		});
	};

	renderNav() {
		const { routes } = this.props;
		return routes.map((link, index) => {
			if (link.children) {
				return (
					<NavGroup key={link.title + index} title={link.title}>
						{this.renderLinks(link.children)}
					</NavGroup>
				);
			} else {
				return this.renderLinks([link]);
			}
		});
	}

	render() {
		return (
			<Styled.Nav
				className={'nav_left' + (this.props.ui && this.props.ui.nav.opened ? '' : ' closed')}
			>
				{this.renderNav()}
			</Styled.Nav>
		);
	}
}

const mapStateToProps = state => {
	return {
		ui: state.ui || {},
	};
};

export default connect(mapStateToProps)(Nav);
