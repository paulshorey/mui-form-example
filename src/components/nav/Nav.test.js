import React from 'react';
import renderer from 'react-test-renderer';

// shallow + redux
import { shallow, connected, shallowConnected } from 'index.testing';
import { Nav } from './Nav';

it('renders correctly', () => {
	const tree = renderer
		.create(
			shallowConnected(
				<Nav
					routes={[
						{
							title: 'Status',
							children: [
								{
									title: 'Controller',
									url: '/XWC1001/status/controller',
									render: <div />,
								},
							],
						},
					]}
				/>
			)
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
