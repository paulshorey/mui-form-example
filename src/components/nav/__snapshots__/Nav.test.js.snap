// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav_left closed sc-bdVaJa fbOaeU"
>
  <div
    className="group"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <span
        className="fontIcon icon-navlink_plus"
      />
      <span
        className="fontIcon icon-navlink_minus"
      />
      <span>
        Status
      </span>
    </div>
    <div
      className="children"
    >
      <a
        aria-current={false}
        className="title link"
        href="/XWC1001/status/controller"
        onClick={[Function]}
        style={undefined}
      >
        <span
          className="fontIcon icon-navlink_dot"
        />
        <span>
          Controller
        </span>
      </a>
    </div>
  </div>
</div>
`;
